rfcs/websockets_http2.md

```diff
@@ -0,0 +1,24 @@
+# RFC #59: Add WebSockets over HTTP/2 support to wptserve
+
+## Summary
+
+Enable wptserve to support WebSockets over an existing http2 connection.
+
+## Details
+
+Allow tests to open a WebSocket over an existing http2 connection.
+
+wptserver needs to implement the [Bootstrapping WebSocket with HTTP/2 protocol](https://tools.ietf.org/html/rfc8441).
+
+It means that it needs to be modified to:
+- Set ENABLE_CONNECT_PROTOCOL settings to 1
+- Detect Extended CONNECT Method and implement the WebSocket opening handshake.
+- Wrap H2Response and H2Request into a Request-like object that pywebsocket can operate on.
+
+## Risks
+
+Most WebSocket tests have a wss variant. We will need to add a new variant in order to use the http2 port when opening the WebSockets.
+
+No other risks are expected except those introduced by any feature e.g. complexity.
+
+This change does not introduce any new dependency.
```

Comment:
But it requires updating some existing dependencies, right? If so, we should list them here just in case.